File: Algebra_SubtractFraction.cs
﻿using System;
using System.Collections.Generic;

namespace Algebra_CS
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Welcome to our demonstration sequels");
            Console.WriteLine("Hope you enjoy (and follow) the lessons.");
            Console.WriteLine("\r\n");

            /*
            * Subtracting fractions
             */
            List<int> numerators = new List<int>();
            numerators.Add(9);
            numerators.Add(3);
            numerators.Add(5);
            numerators.Add(7);

            List<int> denominators = new List<int>();
            denominators.Add(2);
            denominators.Add(4);
            denominators.Add(12);
            denominators.Add(18);

            Console.WriteLine("    Solving:");
            // Print as fraction
            foreach (int n in numerators)
            {
                Console.Write(String.Format("{0,13}", n));
            }
            Console.Write(Environment.NewLine + String.Format("{0,12}", " "));
            for (int i = 0; i < numerators.Count - 1; i++)
            {
                Console.Write(String.Format("{0}", "-     -      "));
            }
            Console.WriteLine(String.Format("{0,1}", "-"));
            foreach (int d in denominators)
            {
                Console.Write(String.Format("{0,13}", d));
            }
            Console.WriteLine();

            // use the SubtractFraction class
            SubtractFraction sub_fract = new SubtractFraction(numerators, denominators);
            int[] solution = sub_fract.doSubtract();

            Console.WriteLine(Environment.NewLine);

            Console.WriteLine(String.Format("{0,25}", solution[0]));
            Console.WriteLine(String.Format("{0,25}", "Answer   =   -"));
            Console.WriteLine(String.Format("{0,25}", solution[1]));

        }
    }
}

File: Algebra_SubtractFraction.pl
#!/usr/bin/perl;
use strict;
use warnings;
use SUBTRACTFRACTION;

# Useful variables
my (@numerators, @denominators, @solutions);
my %fractions;

##
 # Subtracting fractions
 ##
@numerators = (9, 3, 5, 7);
@denominators = (2, 4, 12, 18);
%fractions = (
	numerators	 => \@numerators,
	denominators => \@denominators
);

print "\n    Solving:\n";
# Print as fraction
printf("%13u", $_) for @numerators;
printf("\n%12s", " ");
print "-     -      " for 1 .. $#numerators;
printf("%2s\n", "-");
printf("%13u", $_) for @denominators;
print "\n";

# use the SubtractFraction class
my $sub_fract = SUBTRACTFRACTION->new(\%fractions);
@solutions = $sub_fract->doSubtract();

printf("\n%25u\n", $solutions[0]);
printf("%25s\n", "Answer =     -");
printf("%25u\n", $solutions[1]);


print "\n\n";